cts-java/src/test/java/org/ga4gh/cts/api/references/ReferenceSetsPagingIT.java

```diff
@@ -0,0 +1,132 @@
+package org.ga4gh.cts.api.references;
+
+import org.apache.avro.AvroRemoteException;
+import org.ga4gh.ctk.transport.URLMAPPING;
+import org.ga4gh.ctk.transport.protocols.Client;
+import org.ga4gh.cts.api.Utils;
+import org.ga4gh.cts.api.datasets.DatasetsTests;
+import org.ga4gh.methods.GAException;
+import org.ga4gh.methods.SearchReferenceSetsRequest;
+import org.ga4gh.methods.SearchReferenceSetsResponse;
+import org.ga4gh.models.ReferenceSet;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Test paging through the results returned by <tt>/references/search</tt>.
+ *
+ * @author Herb Jellinek
+ */
+@Category(DatasetsTests.class)
+public class ReferenceSetsPagingIT {
+
+    private static Client client = new Client(URLMAPPING.getInstance());
+
+    /**
+     * Check that we can page 1 by 1 through the {@link ReferenceSet}s we receive from
+     * {@link Client.References#searchReferenceSets(SearchReferenceSetsRequest)}.
+     *
+     * @throws AvroRemoteException if there's a communication problem or server exception ({@link GAException})
+     */
+    @Test
+    public void checkPagingOneByOneThroughReferenceSets() throws AvroRemoteException {
+
+        // retrieve them all
+        final List<ReferenceSet> listOfReferenceSets = Utils.getAllReferenceSets(client);
+        assertThat(listOfReferenceSets).isNotEmpty();
+
+        // we will remove ReferenceSets from this Set and assert at the end that we have zero
+        final Set<ReferenceSet> setOfReferenceSets = new HashSet<>(listOfReferenceSets);
+        assertThat(listOfReferenceSets).hasSize(setOfReferenceSets.size());
+
+        // page through the ReferenceSets using the same query parameters
+        String pageToken = null;
+        for (ReferenceSet ignored : listOfReferenceSets) {
```

Comment:
This pagination pattern is repeated in a few places. Consider the test output when the endpoint returns fewer results than expected. This will result in having us loop back around to the start and emitting confusing assertion failures.

Perhaps accumulate all results into a set and perform a set difference at the end. Or else, assert that pageToken is non-null if there are still more expected reference sets.